In [66]:
## import
import pandas as pd
from pathlib import Path

# import streamlit as st
import plotly.express as px
import plotly.graph_objects as go


TARGET_DIR = 'data'
TARGET_CSV = 'data.csv'

# BASE_DIR = Path(__file__).resolve().parent
BASE_DIR = Path.cwd().resolve().parent
DATA_PATH = BASE_DIR / TARGET_DIR / TARGET_CSV

## read_csv / file: "한국해양진흥공사_부산항_노선별_항만물류지표_KPLI.csv"
df = pd.read_csv(DATA_PATH)


df

,출발항코드(DEPARTURE_PORT_CODE),출발항명(DEPARTURE_PORT_NAME),도착항코드(DEST_PORT_CODE),도착항명(DEST_PORT_NAME),도착항국가(DEST_COUNTRY),기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
0,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-05-28,23.69,644.50,0.00,7.94
1,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-11,22.01,608.54,0.00,16.20
2,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-11,16.52,749.45,0.00,1.36
3,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-30,22.37,634.28,0.00,7.44
4,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-30,16.37,654.99,0.00,1.80
...,...,...,...,...,...,...,...,...,...,...
475,KRPUS,부산항,VNSGN,호치민항,VN,2026-05-29,17.90,158.42,0.16,11.23
476,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,17.12,216.15,0.18,31.91
477,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,17.32,237.27,0.05,7.17
478,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,18.19,232.12,0.02,22.30


# 기초데이터 분석

In [67]:
# 데이터 구조 확인
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 480 entries, 0 to 479
Data columns (total 10 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   출발항코드(DEPARTURE_PORT_CODE)  480 non-null    str    
 1   출발항명(DEPARTURE_PORT_NAME)   480 non-null    str    
 2   도착항코드(DEST_PORT_CODE)       480 non-null    str    
 3   도착항명(DEST_PORT_NAME)        480 non-null    str    
 4   도착항국가(DEST_COUNTRY)         426 non-null    str    
 5   기준일자(DATE)                  480 non-null    str    
 6   항만효율성(PORT_EFFICIENCY)      480 non-null    float64
 7   총항해시간(TOTAL_SAILING_TIME)   480 non-null    float64
 8   대기시간(WAITING_TIME)          480 non-null    float64
 9   항만정시성(ON_TIME_PERFORMANCE)  480 non-null    float64
dtypes: float64(4), str(6)
memory usage: 59.8 KB


In [68]:
# 데이터 모양 확인
df.shape

(480, 10)

In [69]:
# 열 이름 확인
df.columns.to_list()

['출발항코드(DEPARTURE_PORT_CODE)',
 '출발항명(DEPARTURE_PORT_NAME)',
 '도착항코드(DEST_PORT_CODE)',
 '도착항명(DEST_PORT_NAME)',
 '도착항국가(DEST_COUNTRY)',
 '기준일자(DATE)',
 '항만효율성(PORT_EFFICIENCY)',
 '총항해시간(TOTAL_SAILING_TIME)',
 '대기시간(WAITING_TIME)',
 '항만정시성(ON_TIME_PERFORMANCE)']

In [70]:
# 데이터 기초통계
df.describe()

,항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
count,480.000000,480.000000,480.000000,480.000000
mean,19.756458,432.292521,0.138042,6.431250
std,5.754084,415.966180,0.269674,25.602098
min,6.950000,28.970000,0.000000,-322.050000
25%,16.350000,75.985000,0.000000,1.940000
50%,18.735000,327.210000,0.030000,7.645000
75%,21.427500,624.942500,0.170000,13.950000
max,47.010000,1632.540000,3.040000,135.920000


In [71]:
# 결측치 확인
df.isna().sum()

출발항코드(DEPARTURE_PORT_CODE)     0
출발항명(DEPARTURE_PORT_NAME)      0
도착항코드(DEST_PORT_CODE)          0
도착항명(DEST_PORT_NAME)           0
도착항국가(DEST_COUNTRY)           54
기준일자(DATE)                     0
항만효율성(PORT_EFFICIENCY)         0
총항해시간(TOTAL_SAILING_TIME)      0
대기시간(WAITING_TIME)             0
항만정시성(ON_TIME_PERFORMANCE)     0
dtype: int64

상단의 데이터형식을 확인해본 결과(df.info()), 실수형 자료형으로 출력되는 것을 확인.
이를 통해 결측치가 있을 수 있다는 판단 하에 결측치 확인을 수행하였으며,
'도착항국가'가 결측치인 데이터를 발견.

In [72]:
df[df['도착항국가(DEST_COUNTRY)'].isna()]

,출발항코드(DEPARTURE_PORT_CODE),출발항명(DEPARTURE_PORT_NAME),도착항코드(DEST_PORT_CODE),도착항명(DEST_PORT_NAME),도착항국가(DEST_COUNTRY),기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
207,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-05-28,18.99,239.37,0.13,2.29
208,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-06-11,21.48,280.38,0.06,10.84
209,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-06-11,20.45,446.38,0.00,23.58
210,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-06-30,20.06,324.43,0.19,5.72
211,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-06-30,20.67,319.09,0.11,9.72
212,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-06-30,19.70,299.25,0.07,13.01
213,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-07-08,21.16,361.97,0.12,4.02
214,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-07-16,19.06,251.70,0.11,0.90
215,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-07-29,18.63,232.41,0.15,3.09
216,KRPUS,부산항,KPLI,KOBC PORT&LOGISTICS INDICATOR (종합),NaN,2025-07-29,19.45,326.72,0.07,7.87


데이터를 확인해보니 종합 데이터가 추출됨.
이를 통해 해당 파일은 종합데이터와 개별데이터가 있는 것으로 확인됨.

In [73]:
# 날짜 데이터 형식 변경
df['기준일자(DATE)'] = pd.to_datetime(df['기준일자(DATE)'])

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 480 entries, 0 to 479
Data columns (total 10 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   출발항코드(DEPARTURE_PORT_CODE)  480 non-null    str           
 1   출발항명(DEPARTURE_PORT_NAME)   480 non-null    str           
 2   도착항코드(DEST_PORT_CODE)       480 non-null    str           
 3   도착항명(DEST_PORT_NAME)        480 non-null    str           
 4   도착항국가(DEST_COUNTRY)         426 non-null    str           
 5   기준일자(DATE)                  480 non-null    datetime64[us]
 6   항만효율성(PORT_EFFICIENCY)      480 non-null    float64       
 7   총항해시간(TOTAL_SAILING_TIME)   480 non-null    float64       
 8   대기시간(WAITING_TIME)          480 non-null    float64       
 9   항만정시성(ON_TIME_PERFORMANCE)  480 non-null    float64       
dtypes: datetime64[us](1), float64(4), str(5)
memory usage: 55.2 KB


In [74]:
# 기간 확인
df['기준일자(DATE)'].min(), df['기준일자(DATE)'].max()

(Timestamp('2025-05-28 00:00:00'), Timestamp('2026-06-22 00:00:00'))

In [90]:
df['기준일자(DATE)'].unique()

<DatetimeArray>
['2025-05-28 00:00:00', '2025-06-11 00:00:00', '2025-06-30 00:00:00',
 '2025-07-08 00:00:00', '2025-07-29 00:00:00', '2025-08-06 00:00:00',
 '2025-08-13 00:00:00', '2025-08-21 00:00:00', '2025-08-28 00:00:00',
 '2025-09-10 00:00:00', '2025-09-24 00:00:00', '2025-10-15 00:00:00',
 '2025-10-22 00:00:00', '2025-10-28 00:00:00', '2025-11-05 00:00:00',
 '2025-12-02 00:00:00', '2025-12-08 00:00:00', '2025-12-15 00:00:00',
 '2025-12-22 00:00:00', '2025-12-29 00:00:00', '2026-01-05 00:00:00',
 '2026-01-12 00:00:00', '2026-01-19 00:00:00', '2026-01-27 00:00:00',
 '2026-02-02 00:00:00', '2026-02-09 00:00:00', '2026-02-23 00:00:00',
 '2026-03-09 00:00:00', '2025-11-18 00:00:00', '2026-03-30 00:00:00',
 '2026-04-13 00:00:00', '2026-04-24 00:00:00', '2026-06-22 00:00:00',
 '2025-07-16 00:00:00', '2026-03-16 00:00:00', '2026-03-23 00:00:00',
 '2026-04-27 00:00:00', '2026-05-12 00:00:00', '2026-05-29 00:00:00',
 '2026-06-15 00:00:00']
Length: 40, dtype: datetime64[us]

데이터상 시작일자는 25년 5월 28일, 종료일자는 26년 6월 22일. 1년간 데이터 중 40일 데이터를 가져온 것으로 확인됨.

In [76]:
# 일자별 빈도 확인
df['기준일자(DATE)'].value_counts().sort_index(ascending=True)

기준일자(DATE)
2025-05-28    10
2025-06-11    14
2025-06-30    30
2025-07-08    11
2025-07-16     7
2025-07-29    18
2025-08-06    10
2025-08-13    10
2025-08-21    10
2025-08-28     9
2025-09-10    19
2025-09-24    10
2025-10-15    33
2025-10-22     8
2025-10-28     6
2025-11-05    11
2025-11-18    17
2025-12-02     8
2025-12-08     9
2025-12-15    10
2025-12-22     9
2025-12-29     8
2026-01-05    10
2026-01-12    10
2026-01-19     6
2026-01-27     8
2026-02-02     8
2026-02-09    11
2026-02-23    19
2026-03-09    20
2026-03-16     7
2026-03-23     9
2026-03-30    10
2026-04-13    19
2026-04-24     8
2026-04-27     8
2026-05-12     7
2026-05-29     8
2026-06-15    25
2026-06-22    10
Name: count, dtype: int64

추가적으로 각 일자별 건수 또한 균등하지않음.

해당 데이터를 통해 단순 평균을 사용할 경우 특정 일자, 특정 노선의 영향이 커질 수 있으므로, 월평균 및 전체 평균을 기간 전체의 대표값으로 해석하는 데 한계가 있음.
따라서 이후 분석에서는 월평균을 전체 기간의 대표값으로 단정하기보다, 관측된 데이터 범위 내에서 노선별 차이와 시점별 변화 추이를 중심으로 분석.

In [77]:
# 중복데이터 확인
df.duplicated().sum()

np.int64(0)

확인 결과 모든 정보가 동일한 데이터는 확인되지 않았음.

In [78]:
df.duplicated(subset=['기준일자(DATE)','도착항코드(DEST_PORT_CODE)']).sum()

np.int64(106)

하지만 기준일자와 도착항코드만 확인했을때 중복되는 코드가 106개인 것으로 확인됨.

In [79]:
df[
    (df['기준일자(DATE)'] == '2025-10-15')
    &
    (df['도착항코드(DEST_PORT_CODE)'] != 'KPLI')
]

,출발항코드(DEPARTURE_PORT_CODE),출발항명(DEPARTURE_PORT_NAME),도착항코드(DEST_PORT_CODE),도착항명(DEST_PORT_NAME),도착항국가(DEST_COUNTRY),기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
16,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-10-15,21.25,669.18,0.00,1.62
17,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-10-15,16.29,655.12,0.00,1.43
18,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-10-15,25.14,680.22,0.00,4.82
19,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-10-15,29.40,453.14,0.00,4.34
76,KRPUS,부산항,CNSHG,상하이항,CN,2025-10-15,17.18,46.97,0.13,8.54
77,KRPUS,부산항,CNSHG,상하이항,CN,2025-10-15,13.92,41.67,0.21,10.21
78,KRPUS,부산항,CNSHG,상하이항,CN,2025-10-15,17.15,34.61,0.18,-2.31
79,KRPUS,부산항,CNSHG,상하이항,CN,2025-10-15,13.42,37.17,0.22,6.72
128,KRPUS,부산항,CNTAO,칭다오항,CN,2025-10-15,13.23,33.60,0.42,-1.31
129,KRPUS,부산항,CNTAO,칭다오항,CN,2025-10-15,13.68,43.83,0.45,1.08


임의의 데이터를 추출해보니 동일일자 동일도착항의 데이터가 다름.

중복데이터 확인 결과 실제로 중복된 데이터는 관측되지 않았으나(다시확인필요)
동일일자, 동일목적항에 대해 4개의 각기 다른 데이터(항만효율성, 총항해시간, 대기시간, 항만정시성)가 있는 것으로 확인되어
해당 데이터는 관측일자 기준 선박 데이터를 임의로 추출한 것이라고 유추할 수 있음.


In [80]:
# 기초 기술통계
df.describe()

,기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
count,480,480.000000,480.000000,480.000000,480.000000
mean,2025-12-04 05:57:00,19.756458,432.292521,0.138042,6.431250
min,2025-05-28 00:00:00,6.950000,28.970000,0.000000,-322.050000
25%,2025-08-26 06:00:00,16.350000,75.985000,0.000000,1.940000
50%,2025-12-02 00:00:00,18.735000,327.210000,0.030000,7.645000
75%,2026-03-09 00:00:00,21.427500,624.942500,0.170000,13.950000
max,2026-06-22 00:00:00,47.010000,1632.540000,3.040000,135.920000
std,NaN,5.754084,415.966180,0.269674,25.602098


In [81]:
# 종합 데이터와 항만별 데이터 구분.
country_df = df[df['도착항코드(DEST_PORT_CODE)']!= 'KPLI']
kpli_df = df[df['도착항코드(DEST_PORT_CODE)']== 'KPLI']
country_df

,출발항코드(DEPARTURE_PORT_CODE),출발항명(DEPARTURE_PORT_NAME),도착항코드(DEST_PORT_CODE),도착항명(DEST_PORT_NAME),도착항국가(DEST_COUNTRY),기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
0,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-05-28,23.69,644.50,0.00,7.94
1,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-11,22.01,608.54,0.00,16.20
2,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-11,16.52,749.45,0.00,1.36
3,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-30,22.37,634.28,0.00,7.44
4,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-30,16.37,654.99,0.00,1.80
...,...,...,...,...,...,...,...,...,...,...
475,KRPUS,부산항,VNSGN,호치민항,VN,2026-05-29,17.90,158.42,0.16,11.23
476,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,17.12,216.15,0.18,31.91
477,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,17.32,237.27,0.05,7.17
478,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,18.19,232.12,0.02,22.30


In [105]:
country_df.describe()

,기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
count,426,426.000000,426.000000,426.000000,426.000000
mean,2025-12-04 00:13:31.267605,19.878568,446.747887,0.140446,6.165305
min,2025-05-28 00:00:00,6.950000,28.970000,0.000000,-322.050000
25%,2025-08-22 18:00:00,16.070000,71.182500,0.000000,1.810000
50%,2025-12-02 00:00:00,18.625000,331.855000,0.015000,7.210000
75%,2026-03-09 00:00:00,21.760000,658.945000,0.187500,14.077500
max,2026-06-22 00:00:00,47.010000,1632.540000,3.040000,135.920000
std,NaN,6.083060,439.017270,0.285386,26.956670


최대값이 크다는 것을 확인하였으며,
나라별로 차이가 있을 것으로 추정되어 나라별로 구분 후 다시 기술통계를 해야 할 것으로 판단됨.

In [93]:
country_df.groupby('도착항명(country_PORT_NAME)').describe().T

도착항명(DEST_PORT_NAME)                                     뉴욕항  \
기준일자(DATE)                 count                          35   
                           mean   2026-01-03 08:54:51.428571   
                           min           2025-06-11 00:00:00   
                           25%           2025-10-15 00:00:00   
                           50%           2025-12-22 00:00:00   
                           75%           2026-04-06 00:00:00   
                           max           2026-06-22 00:00:00   
                           std                           NaN   
항만효율성(PORT_EFFICIENCY)     count                        35.0   
                           mean                    21.798286   
                           min                         10.95   
                           25%                        19.235   
                           50%                         21.97   
                           75%                        24.795   
                           max                         30.68   
                           std                      4.376728   
총항해시간(TOTAL_SAILING_TIME)  count                        35.0   
                           mean                   717.627429   
                           min                        599.55   
                           25%                        684.63   
                           50%                        720.43   
                           75%                        751.34   
                           max                        872.87   
                           std                     66.316494   
대기시간(WAITING_TIME)         count                        35.0   
                           mean                     0.031143   
                           min                           0.0   
                           25%                           0.0   
                           50%                           0.0   
                           75%                           0.0   
                           max                          0.44   
                           std                      0.095325   
항만정시성(ON_TIME_PERFORMANCE) count                        35.0   
                           mean                    17.179143   
                           min                        -41.29   
                           25%                          3.95   
                           50%                         10.77   
                           75%                         23.17   
                           max                        135.92   
                           std                     27.086585   

도착항명(DEST_PORT_NAME)                             램차방항  \
기준일자(DATE)                 count                    3   
                           mean   2025-06-06 08:00:00   
                           min    2025-05-28 00:00:00   
                           25%    2025-06-04 00:00:00   
                           50%    2025-06-11 00:00:00   
                           75%    2025-06-11 00:00:00   
                           max    2025-06-11 00:00:00   
                           std                    NaN   
항만효율성(PORT_EFFICIENCY)     count                  3.0   
                           mean             18.196667   
                           min                  17.54   
                           25%                  17.78   
                           50%                  18.02   
                           75%                 18.525   
                           max                  19.03   
                           std               0.760548   
총항해시간(TOTAL_SAILING_TIME)  count                  3.0   
                           mean            267.626667   
                           min                 194.96   
                           25%                237.955   
                           50%                 280.95   
                           75%                 303.96   
                           max                 326.97   
       

In [115]:
country_df

,출발항코드(DEPARTURE_PORT_CODE),출발항명(DEPARTURE_PORT_NAME),도착항코드(DEST_PORT_CODE),도착항명(DEST_PORT_NAME),도착항국가(DEST_COUNTRY),기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
0,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-05-28,23.69,644.50,0.00,7.94
1,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-11,22.01,608.54,0.00,16.20
2,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-11,16.52,749.45,0.00,1.36
3,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-30,22.37,634.28,0.00,7.44
4,KRPUS,부산항,AEJEA,제벨알리항,AE,2025-06-30,16.37,654.99,0.00,1.80
...,...,...,...,...,...,...,...,...,...,...
475,KRPUS,부산항,VNSGN,호치민항,VN,2026-05-29,17.90,158.42,0.16,11.23
476,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,17.12,216.15,0.18,31.91
477,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,17.32,237.27,0.05,7.17
478,KRPUS,부산항,VNSGN,호치민항,VN,2026-06-15,18.19,232.12,0.02,22.30


In [119]:
country_desc_df = country_df.groupby('도착항명(country_PORT_NAME)').agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
)

country_desc_df

,항만효율성,총항해시간,대기시간,정시성
도착항명(DEST_PORT_NAME),,,,
뉴욕항,21.798286,717.627429,0.031143,20.203714
램차방항,18.196667,267.626667,0.033333,3.686667
로스앤젤레스항,18.337885,351.327500,0.011154,17.369808
로테르담항,31.964800,1335.029200,0.108000,14.840800
상하이항,15.306731,40.304615,0.246346,11.955385
싱가포르항,20.343704,369.588889,0.031296,15.510926
안트베르펜항,27.923478,1332.163913,0.135652,19.524783
제벨알리항,19.531351,647.047297,0.000000,4.984054
칭다오항,17.289388,39.991020,0.403878,16.634490


항만정시성의 경우 빨리 도착하는 경우도 있기 때문에 해당 데이터는 절대값 처리를 해줬음.
정시에 도착하는것이 좋다. 즉, 0에 가까운 값이 효율이 좋다고 할 수 있음.

In [120]:
country_df.describe().loc['mean']

기준일자(DATE)                    2025-12-04 00:13:31.267605
항만효율성(PORT_EFFICIENCY)                         19.878568
총항해시간(TOTAL_SAILING_TIME)                     446.747887
대기시간(WAITING_TIME)                              0.140446
항만정시성(ON_TIME_PERFORMANCE)                      6.165305
Name: mean, dtype: object

In [121]:
kpli_df.describe().loc['mean']

기준일자(DATE)                    2025-12-06 03:06:40
항만효율성(PORT_EFFICIENCY)                  18.793148
총항해시간(TOTAL_SAILING_TIME)              318.255741
대기시간(WAITING_TIME)                       0.119074
항만정시성(ON_TIME_PERFORMANCE)               8.529259
Name: mean, dtype: object

추가적으로 종합 데이터는 현재 추출된 개별 데이터와 차이가 있어 해당 데이터를 쓰지않고 개별 데이터를 종합하여 따로 구하였음.

In [122]:
overall_mean = pd.Series({
    '항만효율성': country_df['항만효율성(PORT_EFFICIENCY)'].mean(),
    '총항해시간': country_df['총항해시간(TOTAL_SAILING_TIME)'].mean(),
    '대기시간': country_df['대기시간(WAITING_TIME)'].mean(),
    '정시성': country_df['항만정시성(ON_TIME_PERFORMANCE)'].abs().mean()
})
country_desc_df.loc['전체'] = overall_mean

country_desc_df

,항만효율성,총항해시간,대기시간,정시성
도착항명(DEST_PORT_NAME),,,,
뉴욕항,21.798286,717.627429,0.031143,20.203714
램차방항,18.196667,267.626667,0.033333,3.686667
로스앤젤레스항,18.337885,351.327500,0.011154,17.369808
로테르담항,31.964800,1335.029200,0.108000,14.840800
상하이항,15.306731,40.304615,0.246346,11.955385
싱가포르항,20.343704,369.588889,0.031296,15.510926
안트베르펜항,27.923478,1332.163913,0.135652,19.524783
제벨알리항,19.531351,647.047297,0.000000,4.984054
칭다오항,17.289388,39.991020,0.403878,16.634490


# **종합데이터 vs 각 항만 개별데이터**


In [142]:
kpli_df.groupby(pd.to_datetime(kpli_df['기준일자(DATE)']).dt.to_period('M')).agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    항만정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
)

,항만효율성,총항해시간,대기시간,항만정시성
기준일자(DATE),,,,
2025-05,18.990000,239.370000,0.130000,2.290000
2025-06,20.472000,333.906000,0.086000,12.574000
2025-07,19.575000,293.200000,0.112500,3.970000
2025-08,18.922500,336.622500,0.145000,6.217500
2025-09,17.853333,308.896667,0.140000,9.540000
2025-10,18.001667,317.781667,0.145000,10.461667
2025-11,17.930000,336.493333,0.153333,7.263333
2025-12,18.690000,326.548000,0.080000,14.276000
2026-01,19.317500,334.735000,0.150000,11.022500


In [141]:
country_df.groupby(pd.to_datetime(country_df['기준일자(DATE)']).dt.to_period('M')).agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    항만정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
)

,항만효율성,총항해시간,대기시간,항만정시성
기준일자(DATE),,,,
2025-05,18.211111,341.750000,0.123333,5.446667
2025-06,20.884103,478.216154,0.085897,14.722821
2025-07,21.425625,440.201875,0.134687,11.233750
2025-08,20.710286,533.824000,0.165143,11.395143
2025-09,18.349231,493.987692,0.116538,12.016923
2025-10,18.570000,348.740488,0.180244,15.082683
2025-11,19.877200,461.694400,0.177600,15.378800
2025-12,19.583846,468.152564,0.074872,13.669744
2026-01,20.900667,430.941000,0.277667,10.911667


월별 데이터를 평균내어 확인해보았으나 해당 데이터는 맞지 않았음.
아까 확인했던 월별 데이터 개수가 다르므로 일평균을 먼저 내고 월평균을 구하는 것으로 다시 데이터를 확인해보았음.

In [140]:
country_df.groupby(pd.to_datetime(country_df['기준일자(DATE)']).dt.to_period('M')).agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    항만정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
).reset_index()

,기준일자(DATE),항만효율성,총항해시간,대기시간,항만정시성
0,2025-05,18.211111,341.750000,0.123333,5.446667
1,2025-06,20.884103,478.216154,0.085897,14.722821
2,2025-07,21.425625,440.201875,0.134687,11.233750
3,2025-08,20.710286,533.824000,0.165143,11.395143
4,2025-09,18.349231,493.987692,0.116538,12.016923
5,2025-10,18.570000,348.740488,0.180244,15.082683
6,2025-11,19.877200,461.694400,0.177600,15.378800
7,2025-12,19.583846,468.152564,0.074872,13.669744
8,2026-01,20.900667,430.941000,0.277667,10.911667
9,2026-02,19.601176,484.547647,0.089118,19.686765


In [135]:
# 종합의 일 평균
kpli_mean_day_dt = kpli_df.groupby('기준일자(DATE)').agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    항만정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
).reset_index()

In [136]:
# 개별의 일 평균
country_mean_day_dt = country_df.groupby('기준일자(DATE)').agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    항만정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
).reset_index()

In [137]:
kpli_mean_mon_dt = kpli_mean_day_dt.groupby(
    pd.to_datetime(kpli_mean_day_dt['기준일자(DATE)']).dt.to_period('M')
)[['항만효율성',
 '총항해시간',
 '대기시간',
 '항만정시성']].mean()

kpli_mean_mon_dt

,항만효율성,총항해시간,대기시간,항만정시성
기준일자(DATE),,,,
2025-05,18.990000,239.370000,0.130000,2.290000
2025-06,20.554167,338.818333,0.076667,13.346667
2025-07,19.753333,297.745000,0.113333,3.466667
2025-08,18.922500,336.622500,0.145000,6.217500
2025-09,18.047500,318.112500,0.145000,9.135000
2025-10,18.553333,303.633333,0.162500,10.323333
2025-11,18.075000,350.335000,0.125000,7.350000
2025-12,18.690000,326.548000,0.080000,14.276000
2026-01,19.317500,334.735000,0.150000,11.022500


In [139]:
country_mean_mon_dt = country_mean_day_dt.groupby(
    pd.to_datetime(country_mean_day_dt['기준일자(DATE)']).dt.to_period('M')
)[['항만효율성',
 '총항해시간',
 '대기시간',
 '항만정시성']].mean()

country_mean_mon_dt

,항만효율성,총항해시간,대기시간,항만정시성
기준일자(DATE),,,,
2025-05,18.211111,341.750000,0.123333,5.446667
2025-06,20.672546,458.231111,0.069213,14.691944
2025-07,21.098389,419.920667,0.138167,12.538931
2025-08,20.645868,529.333229,0.164028,11.344132
2025-09,18.565621,505.736732,0.114216,12.160915
2025-10,19.041678,338.849117,0.210847,20.558539
2025-11,19.828333,467.999833,0.162833,14.200667
2025-12,19.622262,466.461310,0.074802,13.599079
2026-01,20.396508,420.298135,0.267659,11.522770


평균값이 여전히 다르다.


일별 평균을 구한 이후 월별 평균을 구함. ->
그래도 달랐음.


때문에 더 세분화할 필요성을 느낌.  ->
따라서 목적항별 일자별 월별 로 구분하여 평균을 구함.


평균을 구하는 기준을 잘못 잡은게 아닌가



In [146]:
kpli_mean_dp_day_dt = kpli_df.groupby([
    '도착항코드(DEST_PORT_CODE)','기준일자(DATE)']).agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    항만정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
).reset_index()

kpli_mean_dp_mon_dt = kpli_mean_dp_day_dt.groupby([
    '도착항코드(DEST_PORT_CODE)',
    pd.to_datetime(kpli_mean_dp_day_dt['기준일자(DATE)']).dt.to_period('M')
])[['항만효율성',
 '총항해시간',
 '대기시간',
 '항만정시성']].mean().reset_index()

kpli_mean_dmon_dt = kpli_mean_dp_mon_dt.groupby(['기준일자(DATE)']
)[['항만효율성',
 '총항해시간',
 '대기시간',
 '항만정시성']].mean().reset_index()

kpli_mean_dmon_dt

,기준일자(DATE),항만효율성,총항해시간,대기시간,항만정시성
0,2025-05,18.990000,239.370000,0.130000,2.290000
1,2025-06,20.554167,338.818333,0.076667,13.346667
2,2025-07,19.753333,297.745000,0.113333,3.466667
3,2025-08,18.922500,336.622500,0.145000,6.217500
4,2025-09,18.047500,318.112500,0.145000,9.135000
5,2025-10,18.553333,303.633333,0.162500,10.323333
6,2025-11,18.075000,350.335000,0.125000,7.350000
7,2025-12,18.690000,326.548000,0.080000,14.276000
8,2026-01,19.317500,334.735000,0.150000,11.022500
9,2026-02,19.250000,371.660000,0.090000,9.141667


In [148]:
country_mean_dp_day_dt = country_df.groupby([
    '도착항코드(DEST_PORT_CODE)','기준일자(DATE)']).agg(
    항만효율성=('항만효율성(PORT_EFFICIENCY)', 'mean'),
    총항해시간=('총항해시간(TOTAL_SAILING_TIME)', 'mean'),
    대기시간=('대기시간(WAITING_TIME)', 'mean'),
    항만정시성=('항만정시성(ON_TIME_PERFORMANCE)', lambda x: x.abs().mean()),
).reset_index()

country_mean_dp_mon_dt = country_mean_dp_day_dt.groupby([
    '도착항코드(DEST_PORT_CODE)',
    pd.to_datetime(country_mean_dp_day_dt['기준일자(DATE)']).dt.to_period('M')
])[['항만효율성',
 '총항해시간',
 '대기시간',
 '항만정시성']].mean().reset_index()

country_mean_dmon_dt = country_mean_dp_mon_dt.groupby(['기준일자(DATE)']
)[['항만효율성',
 '총항해시간',
 '대기시간',
 '항만정시성']].mean().reset_index()

country_mean_dmon_dt

,기준일자(DATE),항만효율성,총항해시간,대기시간,항만정시성
0,2025-05,18.211111,341.750000,0.123333,5.446667
1,2025-06,21.064931,547.361944,0.069653,16.968194
2,2025-07,23.260758,587.563712,0.153939,13.985606
3,2025-08,20.762424,565.143712,0.159015,10.408939
4,2025-09,18.283250,501.970250,0.099750,11.764750
5,2025-10,19.960741,429.149136,0.297670,20.654090
6,2025-11,20.634000,515.667750,0.143250,12.988500
7,2025-12,20.601955,604.373182,0.060409,13.929136
8,2026-01,22.385500,515.946833,0.379083,12.279083
9,2026-02,21.616061,620.815455,0.072727,22.134697



도착항코드, 일, 월 순으로 평균을 구했으나
기준일자 25년 6월 항만효율성을 기준으로 보았을때



---
| 구분 | 종합 | 개별 |
| --- | --- | --- |
|전체평균|20.472|20.884|
|일->월평균|20.554|20.672|
|도착항->일->월평균|20.554|20.064|
---
로 다르다는 것을 볼 수 있음.


이를 통해 종합의 데이터는 현재 보유한 표본데이터 이외의 데이터까지 합산한 값이거나 다른 기준을 적용한 값이라는 추측이 가능함.
하지만 현재 종합 말고도 각 항만별 데이터 또한 분석해야하기 때문에 종합 데이터는 사용하지 않고 따로 종합하여 사용하는 것으로 함.

# 노선별 항해시간 시각화 및 이상치 확인

In [155]:
# 노선별 항해시간 box-plot 시각화 및 이상치 확인
fig = px.box(
    country_df,
    x='도착항명(DEST_PORT_NAME)',
    y='총항해시간(TOTAL_SAILING_TIME)',
    color='도착항명(DEST_PORT_NAME)',
    points='outliers'
)
fig.show()

# 추가 데이터 시각화 및 이상치 확인

나머지는 연휴동안 확인하겠습니다...

In [157]:
# 노선별 항만효율성 box-plot 시각화 및 이상치 확인
fig = px.box(
    country_df,
    x='도착항명(DEST_PORT_NAME)',
    y='항만효율성(PORT_EFFICIENCY)',
    color='도착항명(DEST_PORT_NAME)',
    points='outliers'
)
fig.show()

In [158]:
# 노선별 대기시간 box-plot 시각화 및 이상치 확인
fig = px.box(
    country_df,
    x='도착항명(DEST_PORT_NAME)',
    y='대기시간(WAITING_TIME)',
    color='도착항명(DEST_PORT_NAME)',
    points='outliers'
)
fig.show()

In [165]:

for port in country_df['도착항명(DEST_PORT_NAME)'].unique():

    port_df = country_df[
        country_df['도착항명(DEST_PORT_NAME)'] == port
    ]

    q1 = port_df['대기시간(WAITING_TIME)'].quantile(0.25)
    q3 = port_df['대기시간(WAITING_TIME)'].quantile(0.75)

    iqr = q3 - q1

    lower = q1 - 1.5 * iqr
    upper = q3 + 1.5 * iqr

    # port_df['lower'] = lower
    # port_df['upper'] = upper
    print(port, lower, upper)


제벨알리항 0.0 0.0
안트베르펜항 -0.27 0.45
상하이항 -0.13999999999999996 0.6
칭다오항 -0.305 1.0150000000000001
함부르크항 -1.2974999999999999 2.1624999999999996
홍콩항 0.0 0.0
로테르담항 -0.07500000000000001 0.125
싱가포르항 -0.06875 0.12125000000000001
램차방항 -0.0625 0.1175
로스앤젤레스항 0.0 0.0
뉴욕항 0.0 0.0
호치민항 -0.24375000000000005 0.46625000000000005


In [161]:

lower_bound = q1 - 1.5 * iqr
upper_bound = q3 + 1.5 * iqr

outlier = df[
    (df['대기시간(WAITING_TIME)'] < lower_bound)
    |
    (df['대기시간(WAITING_TIME)'] > upper_bound)
]

outlier

,출발항코드(DEPARTURE_PORT_CODE),출발항명(DEPARTURE_PORT_NAME),도착항코드(DEST_PORT_CODE),도착항명(DEST_PORT_NAME),도착항국가(DEST_COUNTRY),기준일자(DATE),항만효율성(PORT_EFFICIENCY),총항해시간(TOTAL_SAILING_TIME),대기시간(WAITING_TIME),항만정시성(ON_TIME_PERFORMANCE)
41,KRPUS,부산항,BEANR,안트베르펜항,BE,2025-08-13,27.51,1123.11,0.62,15.19
47,KRPUS,부산항,BEANR,안트베르펜항,BE,2025-11-05,30.53,1405.51,0.63,-13.57
58,KRPUS,부산항,BEANR,안트베르펜항,BE,2026-04-24,36.72,1385.35,0.70,-40.03
72,KRPUS,부산항,CNSHG,상하이항,CN,2025-08-28,16.80,37.42,0.54,3.06
75,KRPUS,부산항,CNSHG,상하이항,CN,2025-09-24,14.35,42.92,0.49,-5.68
83,KRPUS,부산항,CNSHG,상하이항,CN,2025-11-18,15.80,40.83,0.48,10.74
90,KRPUS,부산항,CNSHG,상하이항,CN,2026-01-05,18.20,32.81,0.47,1.91
96,KRPUS,부산항,CNSHG,상하이항,CN,2026-02-23,14.56,40.91,0.84,7.27
108,KRPUS,부산항,CNSHG,상하이항,CN,2026-06-15,15.80,37.46,0.78,0.96
109,KRPUS,부산항,CNSHG,상하이항,CN,2026-06-15,16.14,40.56,0.56,3.39


In [160]:
# 노선별 항만정시성 box-plot 시각화 및 이상치 확인
fig = px.box(
    country_df,
    x='도착항명(DEST_PORT_NAME)',
    y='항만정시성(ON_TIME_PERFORMANCE)',
    color='도착항명(DEST_PORT_NAME)',
    points='outliers'
)
fig.show()